x/bank/keeper/keeper_test.go

```diff
@@ -1231,6 +1231,357 @@ func (suite *IntegrationTestSuite) TestMintCoinRestrictions() {
 	}
 }
 
+func (suite *IntegrationTestSuite) TestSendRestrictions() {
+	type BankSendRestrictionFn func(ctx sdk.Context, fromAddr, toAddr sdk.AccAddress, amt sdk.Coins) (sdk.AccAddress, error)
+
+	// Define a new module account for the test
+	newModuleAcc := authtypes.NewEmptyModuleAccount("newbankmodule", authtypes.Minter, authtypes.Burner)
+
+	// Setup the permissions for the module account
+	maccPerms := simapp.GetMaccPerms()
+	maccPerms[newModuleAcc.Name] = []string{authtypes.Burner, authtypes.Minter, authtypes.Staking}
+
+	// Update the AccountKeeper with the new module account
+	suite.app.AccountKeeper = authkeeper.NewAccountKeeper(
+		suite.app.AppCodec(), suite.app.GetKey(authtypes.StoreKey), suite.app.GetSubspace(authtypes.ModuleName),
+		authtypes.ProtoBaseAccount, maccPerms,
+	)
+	suite.app.AccountKeeper.SetModuleAccount(suite.ctx, newModuleAcc)
+
+	// Initialize the BankKeeper
+	suite.app.BankKeeper = keeper.NewBaseKeeper(
+		suite.app.AppCodec(), suite.app.GetKey(types.StoreKey),
+		suite.app.AccountKeeper, suite.app.GetSubspace(types.ModuleName), nil,
+	)
+
+	// Define normal addresses
+	normalAddr1 := sdk.AccAddress([]byte("addr1---------------"))
+	normalAddr2 := sdk.AccAddress([]byte("addr2---------------"))
+	blockedAddr := sdk.AccAddress([]byte("blockedaddr---------"))
```

Comment:
Type conversion isn't necessary. The following code also works.
```suggestion
	normalAddr1 := sdk.AccAddress("addr1---------------")
	normalAddr2 := sdk.AccAddress("addr2---------------")
	blockedAddr := sdk.AccAddress("blockedaddr---------")
```

Reply:
This part was written following the existing Cosmos SDK test code convention. In the new test code added for this PR,  I have removed all byte conversions but not existing tests. https://github.com/b-harvest/cosmos-sdk/pull/173/commits/deb55d353d75998ab7451b449c17af3bb1bea2de